src/data/chats/chat.rs

```diff
@@ -35,24 +41,72 @@ enum TitleState {
     Updated,
 }
 
-#[derive(Debug)]
+#[derive(Serialize, Debug)]
 pub struct Chat {
     /// Unix timestamp in ms.
     pub id: ChatID,
     pub model_filename: String,
     pub file_id: FileID,
     pub messages: Vec<ChatMessage>,
+    #[serde(skip)]
     pub messages_update_sender: Sender<ChatTokenArrivalAction>,
+    #[serde(skip)]
     pub messages_update_receiver: Receiver<ChatTokenArrivalAction>,
+    #[serde(skip)]
     pub is_streaming: bool,
 
     title: String,
     /// Know when title was updated by user.
+    #[serde(skip)]
     title_state: TitleState,
+
+    #[serde(skip)]
+    chats_dir: PathBuf,
+}
+
+impl<'de> Deserialize<'de> for Chat {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        struct ChatData {
+            id: ChatID,
+            model_filename: String,
+            file_id: FileID,
+            messages: Vec<ChatMessage>,
+            title: String,
+        }
+
+        let ChatData {
+            id,
+            model_filename,
+            file_id,
+            messages,
+            title,
+        } = ChatData::deserialize(deserializer)?;
+
+        let (tx, rx) = channel();
+
+        Ok(Chat {
+            id,
+            model_filename,
+            file_id,
+            messages,
+            // TODO: This is kinda bad, we set the sender and reciever which need to
+            // be set correctly later
```

Comment:
Looking at this comment I feel we shouldn't have our `Chat` struct implementing the serialization/deserialization to JSON. An alternative is to have functions converting `Chat` to/from `ChatData` (which can implement `Serde` easily), and make use of those functions in `load` and `save` functions. This idea has the benefit of removing all the `serde(skip)` added in the main struct.

In any case, I wouldn't block this PR for this. Just sharing how I would address this comment.